docs/reference/index.rst

```diff
@@ -1,11 +1,62 @@
-=========
-Reference
-=========
+.. _api:
 
+================
 Flocker REST API
 ================
 
-.. toctree::
-   :maxdepth: 2
+Introduction
+============
+
+In general the API allows for modifying the desired configuration of the cluster.
+When you use the API to change the configuration, e.g. creating a new dataset:
+
+You will need to provide API end users with a certificate for authentication before they can use the API.
+For more information, see the :ref:`API authentication guide <generate-api>`.
+
+#. A successful response indicates a change in configuration, not a change to cluster state.
+#. Convergence agents will then take the necessary actions and eventually the cluster's state will match the requested configuration.
+#. The actual cluster state will then reflect the requested change.
+   For example, you can retrieve the current cluster datasets via :http:get:`/v1/state/datasets`.
+
+.. XXX: Document the response when input validation fails:
+.. https://clusterhq.atlassian.net/browse/FLOC-1613
+
+For more information read the :ref:`cluster architecture<architecture>` documentation.
+
+REST API Endpoints
+==================
+
+.. contents::
+        :local:
+
+Common API Endpoints
```

Comment:
If we only have the single "dataset" API, then having a "common" API doesn't make sense. Does it work better to remove the headers for the the two sections?